M2/Macaulay2/m2/ringmap.m2

```diff
@@ -85,7 +83,7 @@ map(Ring, Ring, Matrix)  := RingMap => opts -> (R, S, m) -> (
 		    " into a degree of length ", toString degreeLength R);
 	       opts.DegreeMap
 	       )
-	  else if workable (() -> promote({},S,R)) then (d -> first promote({d},S,R))
+	  else if (pr:=lookup(promote,List,S,R)) =!= null then (d -> first pr({d},S,R))
```

Comment:
Huh, I didn't know `promote(List,S,R)` is for promoting the degree till now ...

Reply:
For the record I fixed this line because it could create infinite loops. It pretty much never occurred before because there were few `promote`, but now with the added flexibility of `promoteFromMap` (or whatever we're going to end up calling), it can easily happen.

Reply:
I think I would be in favor of deprecating `promote(List, Ring, Ring)` and instead given a ring map `f` make use of `f.degreeMap` and `f.degreeLift`.

Reply:
Or maybe just `degreeGroup f` defined as `degreeGroup RingMap := f -> f.degreeMap`, which is similar to how NormalToricVarieties supports `picardGroup ToricMap` to give the map of Picard groups functorially.